### ISSUE TITLE
Proxy IdPs with domain name in ALL CAPS or MixedCase treated as secondary

### ISSUE THREAD

**Issue by karlht** [mozilla/persona#3245]
@jrgm to add colour commentary -- but the basic case is this:

karl_thiessen@yahoo.com is treated as a proxied primary.
karl_thiessen@YAHOO.COM is treated as a secondary.

**Comment by jrgm** [comment]
Here is the difference, between a primary in upper/lower, and a proxy_idp domain in upper/lower.

https://login.anosrep.org/wsapi/address_info?email=somerandom%40eyedee.me

```
{"auth":"https://eyedee.me/browserid/sign_in.html","prov":"https://eyedee.me/browserid/provision.html",
 "type":"primary","issuer":"eyedee.me","state":"unknown"}
```

https://login.anosrep.org/wsapi/address_info?email=somerandom%40EYEDEE.ME

```
{"auth":"https://EYEDEE.ME/browserid/sign_in.html","prov":"https://EYEDEE.ME/browserid/provision.html",
 "type":"primary","issuer":"EYEDEE.ME","state":"unknown"}
```

https://login.anosrep.org/wsapi/address_info?email=somerandom%40yahoo.com

```
{"auth":"https://yahoo.login.anosrep.org/authentication","prov":"https://yahoo.login.anosrep.org/provision",
 "type":"primary","issuer":"yahoo.login.anosrep.org","state":"unknown"}
```

https://login.anosrep.org/wsapi/address_info?email=somerandom%40YAHOO.COM

```
{"type":"secondary","issuer":"login.anosrep.org","disabled":false,"state":"unknown"}
```

I'm guessing the issue is this line, and `currentDomain` comes back in the original case.

https://github.com/mozilla/browserid/blob/dev/lib/primary.js#L100

**Comment by seanmonstar** [comment]
this should be fixed with the same commits that fixed #3728. That Yahoo still doesn't work is a bug in browserid-bigtent.

**Comment by edr67** [comment]
I am still seeing the issue (Gmail.com is not responding...), with the email address I reported (ARJ412@Gmail.com). Is there any date that I can tell this user that his email ID will work in Persona again? It's been a while since he's been able to log in to my website.

As a last resort can somebody delete his persona ID and I can have him start over, and use all lower case in the gmail.com portion of his email address.

Thank You
 Ed

**Comment by jaredhirsch** [comment]
@edr67 we've just released the fix for #3728. If your user is still having problems, please reopen that bug and we'll get right on it.

edit: #3728 is the fix for using 'Gmail' instead of 'gmail' as the domain name.

**Comment by shane-tomlinson** [comment]
If @seanmonstar says this is fixed, closing as fixed. Re-open if STRs are found to reproduce once this makes it into prod.
